### PR TITLE
config: enable-stats-cache-mem-quota is set to enable by default

### LINKED ISSUE #46158: Set enable-stats-cache-mem-quota to default configuration
## Enhancement

Currently, the enable-stats-cache-mem-quota flag is not enabled by default in our system. This flag controls the memory quota for the stats cache, which is an important feature for efficient resource allocation and management.
To improve the usability and performance of our system, it is recommended to set the enable-stats-cache-mem-quota flag to be enabled by default. This will ensure that the stats cache memory is properly allocated and managed, preventing potential memory overflows and improving overall system stability.

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #46158

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: statistics/handle/cache/internal/lfu/lfu_cache.go [hunk 1]
```diff
@@ -149,6 +150,9 @@ func (s *LFU) dropMemory(item *ristretto.Item) {
 		// so it should not be processed.
 		return
 	}
+	if s.closed.Load() {
```
Comment (hawkingrei) [inline statistics/handle/cache/internal/lfu/lfu_cache.go:153]: We will not evict anything when to exit.

File: statistics/handle/handle.go [hunk 2]
```diff
@@ -2292,6 +2292,11 @@ func (h *Handle) SetStatsCacheCapacity(c int64) {
...
 // Close stops the background
 func (h *Handle) Close() {
-	h.statsCache.Load().Close()
 	h.gpool.Close()
+	h.CloseStatsCache()
+}
+
+// CloseStatsCache stops the statsCache
+func (h *Handle) CloseStatsCache() {
+	h.statsCache.Load().Close()
```
Comment (qw4990) [inline statistics/handle/handle.go:2301]: Is this function idempotent?

File: statistics/handle/handletest/handle_test.go [hunk 3]
```diff
@@ -169,6 +169,9 @@ func TestVersion(t *testing.T) {
 	require.NoError(t, err)
 	tableInfo1 := tbl1.Meta()
 	h, err := handle.NewHandle(testKit.Session(), testKit2.Session(), time.Millisecond, do.SysSessionPool(), do.SysProcTracker(), do.GetAutoAnalyzeProcID)
+	defer func() {
+		h.CloseStatsCache()
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply statistics/handle/handletest/handle_test.go]: Done

File: statistics/handle/handle.go (hunk 2 above)
Comment (hawkingrei) [reply statistics/handle/handle.go:2296]: It has been removed.

File: statistics/handle/bootstrap.go [hunk 4]
```diff
@@ -164,6 +164,7 @@ func (h *Handle) initStatsHistograms4Chunk(is infoschema.InfoSchema, cache *cach
 	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
 		tblID, statsVer := row.GetInt64(0), row.GetInt64(8)
 		table, ok := cache.GetFromInternal(tblID)
```
[the remark this answers is not in the document]
Comment (hawkingrei) [reply statistics/handle/bootstrap.go:167]: Done

**Review by qw4990 — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Comment by easonn7** [comment]
/approve

The primary purpose of this "config" is to serve as a safeguard switch to prevent unexpected behavior after the LFU implementation is released. It ensures that the stats cache reverts to the behavior of the map cache. From a PM perspective, the naming of this config has a gap compared to its actual behavior. Personally, I believe that this setting should not be provided. However, considering the security assurance for new features, I will approve this "config". 

In the long term, this "config" should be deprecated, and we should design a variable similar to "fix/feature_control" to manage the toggling behavior of these temporary features. This will avoid adding and deprecating configs/variables for each feature. We will have a separate discussion on this matter.

**Pull request merged**